modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js

```diff
@@ -58,25 +59,37 @@ function normalizeInputValue(fieldType, locale, value) {
 	return value;
 }
 
-const getFieldDetails = ({errorMessage, hasError, required, text, tip}) => {
-	let fieldDetails = '';
+const getFieldDetails = ({
+	errorMessage,
+	hasError,
+	required,
+	text,
+	tip,
+	warningMessage,
+}) => {
+	const fieldDetails = [];
 
 	if (tip) {
-		fieldDetails += Liferay.Util.escape(tip) + '<br>';
+		fieldDetails.push(Liferay.Util.escape(tip));
 	}
 
 	if (text) {
-		fieldDetails += Liferay.Util.escape(text) + '<br>';
+		fieldDetails.push(Liferay.Util.escape(text));
 	}
 
 	if (hasError) {
-		fieldDetails += Liferay.Util.escape(errorMessage);
+		fieldDetails.push(Liferay.Util.escape(errorMessage));
 	}
-	else if (required) {
-		fieldDetails += Liferay.Language.get('required');
+	else {
+		if (warningMessage) {
```

Comment:
On this case, the warning message and the required error will appear at the same time right? Is this the expected behavior?

Reply:
That was a great observation, but in this case if there is an error no warning nor help text will be displayed since it is into the `else` clause of the ` if (hasError)`.

Anyway it is inconsistent with the displayed information where the help text is always present. Since it was the original behaviour I will fix it in a new PR.